distributed/utils_test.py

```diff
@@ -1769,6 +1770,17 @@ def check_instances():
 
     _global_clients.clear()
 
+    for s in Scheduler._instances:
+        s.extensions.clear()
+        s.plugins.clear()
+        s.services.clear()
+        s.handlers.clear()
+        s.stream_handlers.clear()
+        s.stream_comms.clear()
+        # No close methods, destroy them
+        del s.http_application
+        del s.http_server
```

Comment:
This is a WIP. I suspect all this logic should be placed elsewhere (in C++, it would be the destructor). The `close` method might be the closest thing to this.

/cc @fjetter

Reply:
Place in `clear_instance` and `clear_instances` methods